lib/plugins/TaskRunner/Drupal.js

```diff
@@ -0,0 +1,171 @@
+   *   - siteFolder: (string) - The site folder to use for this build (the folder within the drupal `sites` folder).  Defaults to `default`.
+   *   - database: (string) - The name of the database to import if specified. Note that this database *must be added to the assets array separately*.
+   *   - databaseGzipped: (boolean) - Whether the database was sent gzipped and whether it should therefore be gunzipped before importing.
+   *   - databaseUpdates: (boolean) - Determines whether to run `drush updb`.
+   *   - revertFeatures: (boolean) - Whether to revert features using `drush fra`.
+   *   - makeFile: (string) - The name of the make file to run to generate the install directory.
+   *   - installProfile: (string) - If set this install (incompatible with the `database` option).
+   *   - runInstall: (boolean) - If set, run `drush site-install` to perform a fresh install of the site using the profileName as the profile to install and allowing instlallArgs to configure the install.
+   *   - profileName: (string) - The profileName, used in symlinking this directory if makeFile is specified and used to select the profile to install if `runInstall` is selected.
+   *   - installArgs: (string) - A set of params to concat onto the drush `site-install` command (defaults to '').
+   */
+  constructor(container, options) {
+    super(container, options)
+
+    this.options.siteFolder = options.siteFolder || 'default'
+    this.options.profileName = options.profileName || 'standard'
+    this.options.profileName = options.clearCaches != undefined ? options.clearCaches : true
+
+    // TODO: Add some kind of validation.
+
+    // Filter out secret strings
+    options.secrets = [
+    ]
+
+    var subDirectory = options.subDirectory || 'docroot'
+    this.script = []
+    this.populateScriptArray()
+    this.setScript(this.script)
+
+  }
+
+  description(){
+    return `${this.plugin} 'Provisioning Drupal!'`
+  }
+
+  /**
+   *
+   */
+  populateScriptArray() {
+    this.addScriptHeader()
+    if (this.options.makeFile) {
+      this.addScriptRunMakeFile()
+    }
+    else {
+      this.addScriptSymlinks()
+    }
+    this.addScriptCreateDatbase()
+    this.addScriptAppendSettingsPHPSettings()
+    this.addScriptPublicFilesDirectory()
+    if (this.options.database) {
+      this.addScriptImportDatabase()
+    }
+    if (this.options.runInstall) {
+      this.addScriptRunInstall()
+    }
+    if (this.options.databaseUpdates) {
+      this.addScriptDatabaseUpdates()
+    }
+    if (this.options.revertFeatures) {
+      this.addScriptRevertFeatures()
+    }
+    if (this.options.clearCaches) {
+      this.addScriptClearCaches()
+    }
+  }
+
+  addScriptHeader() {
+    this.script = this.script.concat([
+      'sleep 3',
+    ])
+  }
+
+  addScriptSymlinks() {
+    this.script = this.script.concat([
+      'if [ -d "$SRC_DIR/docroot" ] ; then',
```

Comment:
Yeah, but I think wrapping the line as weirdly as that does makes things harder to read.  I personally like the idea of turning the thing into an array and then never needing to keep track of whether we're in "big string" mode or "array mode" - if it comes in as a string, turn it into an array, then turn it back before use.